public/src/inverted-index.js

```diff
@@ -92,63 +106,61 @@ class Index {
     return check;
   }
 
-/**
- * @method getIndex
- *
- * Method return index of files in indices object
- *
- * @param {string} name
- * @returns {object}
- */
+  /**
+  * @method getIndex
+  *
+  * Method return index of files in indices object
+  *
+  * @param {string} name
+  * @returns {object}
+  */
   getIndex(name) {
-    if (name && typeof name === 'string') {
-      return this.indices[name];
-    }
-    return this.indices;
+    return name && typeof name === 'string' ?
+      this.indices[name] : this.indices;
   }
 
-/**
- * @method searchFeedback
- *
- * Takes terms of array and fetch result of each token.
- *
- * @param {array} termsArray
- * @param {object} file
- * @returns {object} searchResults
- */
-  searchFeedback(termsArray, file) {
+
+  /**
+  * @method doSearch
+  *
+  * Takes terms of array and fetch result of each token.
+  *
+  * @param {array} termsArray
+  * @param {object} file
+  * @returns {object} searchResults
+  */
+  doSearch(termsArray, fileIndex) {
     const searchResults = {};
     termsArray.forEach((term, index) => {
-      if (file.hasOwnProperty(term)) {
-        searchResults[termsArray[index]] = Array.from(file[term]);
+      if (fileIndex.hasOwnProperty(term)) {
+        searchResults[termsArray[index]] = Array.from(fileIndex[term]);
       } else {
         searchResults[term] = [];
       }
     });
     return searchResults;
   }
 
-/**
- * @method searchIndex
- *
- * Looks for search terms in created index
- *
- * @param {object} currentFile
- * @param {array} searchTerm
- * @returns {object} searchResults
- */
-  searchIndex(currentFile, ...searchTerm) {
+  /**
+  * @method searchIndex
+  *
+  * Looks for search terms in created index
+  *
+  * @param {object} selectedFile
+  * @param {array} searchTerms
+  * @returns {object} searchResults
+  */
+  searchIndex(selectedFile, ...searchTerms) {
     const searchResults = {};
-    let termsArray = [];
-    termsArray = this.sanitizeInput(searchTerm[0]);
-    if (!currentFile) {
+    const termsArray = this.sanitizeInput(searchTerms[0]);
+    if (!selectedFile) {
```

Comment:
This method is not defaulting to the else block as you commented.Because the requirement was to allow the user to either search from a single file of his/her choice or from all the files.So what is happening here is that in the front-end, there is a select tag which has two options.The first option is searching against all the files of which am passing in a blank/no value.The other option(s) tag(s) is/are dynamic according to the number of indexed files,having the value as the name of the file,for example <code>books.json</code>. If a file is selected,<code>selectedFile</code> holds the name of the file selected.All these are handled by angular.If a file to be searched against is not provided,then in my case the <code>selectedFile</code> becomes <code>undefined</code> because no file name is passed to it,hence search is done against all the files.